File: ch01/fib.r
# 方法1：递归实现
fib_recursive <- function(n) {
  # 基础情况
  if (n <= 1) {
    return(n)
  }
  
  # 递归计算
  return(fib_recursive(n - 1) + fib_recursive(n - 2))
}

# 方法2：迭代实现
fib_iterative <- function(n) {
  # 处理基础情况
  if (n <= 1) {
    return(n)
  }
  
  prev <- 0
  current <- 1
  
  # 迭代计算
  for (i in 2:n) {
    next_val <- prev + current
    prev <- current
    current <- next_val
  }
  
  return(current)
}

# 测试代码
cat("递归方法:\n")
for (i in 0:9) {
  cat(sprintf("F(%d) = %d\n", i, fib_recursive(i)))
}

cat("\n迭代方法:\n")
for (i in 0:9) {
  cat(sprintf("F(%d) = %d\n", i, fib_iterative(i)))
} 

File: ch01/fib.rs
// 方法1：递归实现
fn fib_recursive(n: u32) -> u32 {
    // 基础情况
    if n <= 1 {
        return n;
    }

    // 递归计算
    fib_recursive(n - 1) + fib_recursive(n - 2)
}

// 方法2：迭代实现
fn fib_iterative(n: u32) -> u32 {
    // 处理基础情况
    if n <= 1 {
        return n;
    }

    let mut prev = 0;
    let mut current = 1;

    // 迭代计算
    for _ in 2..=n {
        let next = prev + current;
        prev = current;
        current = next;
    }

    current
}

fn main() {
    println!("递归方法:");
    for i in 0..10 {
        println!("F({}) = {}", i, fib_recursive(i));
    }

    println!("\n迭代方法:");
    for i in 0..10 {
        println!("F({}) = {}", i, fib_iterative(i));
    }
} 